lib/pushy_client/protocol_handler.rb

```diff
@@ -413,10 +406,15 @@ def check_server_address(name, server_address)
     def validate_params(params = {})
       stdout_bytes = params[:stdout].to_s.bytesize
       stderr_bytes = params[:stderr].to_s.bytesize
-
-      if (stdout_bytes + stderr_bytes) > MAX_BODY_SIZE
+      if (stdout_bytes + stderr_bytes) > client.max_body_size.to_i
```

Comment:
As mentioned in line 62 of this file, we are measuring the payload size here. However the value the server is communicating is the max size of the packet it will accept. That's why we hardcoded 63000 rather than 65535 like the server used, to account for the packet overhead (headers, etc)

So we probably should hoist this code to be measuring the full packet size, or alternatively, figure out how much overhead we need and apply it to client.max_body_size.

Reply:
@markan Will subtracting `2535` from the value be an acceptable workaround? If not, @piyushawasthi can probably figure out the packet overhead via wireshark or tcpdump.

Reply:
@piyushawasthi -- @markan Is on vacation, but hopefully he will be able to respond soon.

Reply:
@markan, @rhass - Please review this PR.

Reply:
I think 2535 is overkill. The right thing to do is move the size check to send_signed_json_command https://github.com/chef/opscode-pushy-client/blob/master/lib/pushy_client/protocol_handler.rb#L474
Something like
testing message.to_s.bytesize > client.max_body_size.to_i  and then printing a helpful error message would be clean and stable.

I'm writing a bit of sample code now.

Reply:
See ma/config_max_body_size branch for some ideas.

Reply:
@markan : Please review the changes and let me know if that work for you.